chef/cookbooks/swift/recipes/disks.rb

```diff
@@ -30,12 +30,14 @@ def get_uuid(disk)
   nil
 end
 
-unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node)
-to_use_disks = []
-unclaimed_disks.each do |k|
-  to_use_disks << k
-end
-claimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.claimed(node, "Swift")
+claim_string = "Swift"
+
+# take only disks with Swift role; when there is none, take disks without role
+swift_disks, disks_without_role =
+  BarclampLibrary::Barclamp::Inventory::Disk.prepared_for(node, claim_string)
+to_use_disks = swift_disks.empty? ? disks_without_role : swift_disks
```

Comment:
Hm, does this mean, that in the next chef run (after disk with swift role was already used), disk with no role will be taken anyway? That doesn't seem to help...